daemon/openrazer_daemon/hardware/keyboards.py

```diff
@@ -1113,3 +1113,28 @@ class RazerBlade2019Adv(_RippleKeyboard):
         "side_img": "https://assets2.razerzone.com/images/blade-15/blade15-usp-ports-advanced-model.jpg",
         "perspective_img": "https://d4kkpd69xt9l7.cloudfront.net/sys-master/root/h78/h94/9126618628126/razer-blade-15-gallery07.jpg"
     }
+
+
+class RazerBladeMid2019Mercury(_RippleKeyboard):
+    """
+    Class for the Razer Blade 15 (Mid 2019) Mercury
+    """
+    EVENT_FILE_REGEX = re.compile(r'.*Razer_Blade(-if01)?-event-kbd')
+
+    USB_VID = 0x1532
+    USB_PID = 0x0245
+    HAS_MATRIX = True
+    MATRIX_DIMS = [6, 16]
+    METHODS = ['get_device_type_keyboard', 'set_wave_effect', 'set_static_effect', 'set_spectrum_effect',
+               'set_reactive_effect', 'set_none_effect', 'set_breath_random_effect', 'set_breath_single_effect',
+               'set_breath_dual_effect', 'set_custom_effect', 'set_key_row', 'set_starlight_random_effect',
+               'set_ripple_effect', 'set_ripple_effect_random_colour']
+
+    DEVICE_IMAGE = "https://assets2.razerzone.com/images/blade-15/shop/blade15-mercury-1.jpg"
+
+    # Deprecated - RAZER_URLS be removed in future.
+    RAZER_URLS = {
+        "top_img": "https://assets.razerzone.com/eeimages/support/products/1482/blade15.png",
+        "side_img": "https://assets2.razerzone.com/images/blade-15/blade15-usp-ports-advanced-model.jpg",
+        "perspective_img": "https://d4kkpd69xt9l7.cloudfront.net/sys-master/root/h78/h94/9126618628126/razer-blade-15-gallery07.jpg"
+    }
```

Comment:
Please remove this block

Reply:
Yes, I've got a Blade Advanced Early 2019 Mercury White and all the effects work fine. The Mid 2019 model has just a different pid. 
I'll make the changes you requested soon enough.

Reply:
Done.

Reply:
> Looks good overall, just a small comment.
> 
> Do all effects work with that PR?

I couldn't get my wave effect to work properly. It doesn't have that fade-in/fade-out effect. Also, I couldn't get my keyboard work with different colors for each key. It seems that all keys are stuck with the same color. No independent color changing.